pkg/client/dtclient/settings_client_test.go

```diff
@@ -517,6 +556,22 @@ func Test_findObjectWithSameConstraints(t *testing.T) {
 					},
 				},
 			},
+			{
```

Comment:
I would suggest adding the next error cases to document that for matching the full path needs to be same:

```
					schema: Schema{
						UniqueProperties: [][]string{
							{"B"},
						},
					},
					source: SettingsObject{
						SchemaId: "schemaID", Content: []byte(`{"A" : {"B" : "x" } }`),
					},
					objects: []DownloadSettingsObject{
						{Value: []byte(`{"A" : {"B" : "x" } }`)},
					},
``` 

```
					schema: Schema{
						UniqueProperties: [][]string{
							{"B"},
						},
					},
					source: SettingsObject{
						SchemaId: "schemaID", Content: []byte(`{"B" : "x" } `),
					},
					objects: []DownloadSettingsObject{
						{Value: []byte(`{"A" : {"B" : "x" } }`)},
					},
```

Reply:
thanks, good suggestion :)